index.js

```diff
@@ -56,6 +56,14 @@ function attachComments(text, ast, opts) {
 }
 
 function ensureAllCommentsPrinted(astComments) {
+  for (let i = 0; i < astComments.length; ++i) {
+    if (astComments[i].value.trim() === "prettier-ignore") {
+      // If there's a prettier-ignore, we're not printing that sub-tree so we
+      // don't know if the comments was printed or not.
+      return;
+    }
+  }
+
```

Comment:
This seems like it could have a perf hit. Not a huge one, but certainly doing stuff like this in several places will add up to a non-negligible perf cost.

Do we have a plan for removing this check at some point? We shouldn't need to once we know that we are stable enough.